Bottom: a06cca4299256200b8539c791eaed27c80ae95f2
Top:    7a0faaf88a414d1135f79a323a79aeac3ca8acc4
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-11-09 19:05:31 +0000

Tmp


---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index afd0b86..841f612 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -1618,18 +1618,14 @@ ghid_draw_everything (BoxTypePtr drawn_area)
   /* Test direction of rendering */
   /* Look at sign of eye coordinate system z-coord when projecting a
      world vector along +ve Z axis, (0, 0, 1). */
-  /* FIXME: This isn't strictly correct, as I've ignored the matrix
-            elements for homogeneous coordinates. */
+  /* XXX: This isn't strictly correct, as I've ignored the matrix
+          elements for homogeneous coordinates. */
   /* NB: last_modelview_matrix is transposed in memory! */
   reverse_layers = (last_modelview_matrix[2][2] < 0);
 
   save_show_solder = Settings.ShowSolderSide;
+  Settings.ShowSolderSide = reverse_layers;
 
-  if (reverse_layers)
-    Settings.ShowSolderSide = !Settings.ShowSolderSide;
-
-  if (!global_view_2d && save_show_solder)
-    reverse_layers = !reverse_layers;
   PCB->Data->SILKLAYER.Color = PCB->ElementColor;
   PCB->Data->BACKSILKLAYER.Color = PCB->InvisibleObjectsColor;
